extension/src/time_vector.rs

```diff
@@ -115,6 +119,48 @@ pub fn unnest<'a>(
     )
 }
 
+/// Util function to convert from *const ::std::os::raw::c_char to String
+/// TimestampTz -> *const c_char -> &CStr -> &str -> String
+pub fn timestamptz_to_string(time: pg_sys::TimestampTz) -> Result<String, Utf8Error> {
+    let char_ptr = unsafe { timestamptz_to_str(time) };
+    let c_str = unsafe { CStr::from_ptr(char_ptr) };
+    c_str.to_str().map(|s| s.to_owned())
+}
+
+#[pg_extern(immutable, schema = "toolkit_experimental", parallel_safe)]
+pub fn format_timevector<'a>(series: Timevector_TSTZ_F64<'a>, format_string: String) -> String {
+    let mut context = Context::new();
+    let mut times: Vec<String> = Vec::new();
+    let mut values: Vec<String> = Vec::new();
+    if series.has_nulls() {
+        for (i, point) in series.iter().enumerate() {
+            times.push(timestamptz_to_string(point.ts).unwrap());
+            if series.is_null_val(i) {
+                values.push("null".to_string())
+            } else {
+                match point.val.to_string().as_ref() {
+                    "NaN" | "Inf" => panic!(),
```

Comment:
Could use a better error message here:
```suggestion
                    "NaN" | "Inf" => panic!("All values in the series must be finite"),
```